server/src/main/java/io/crate/replication/logical/action/GetFileChunkAction.java

```diff
@@ -0,0 +1,225 @@
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.routing.ShardsIterator;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.index.store.StoreFileMetadata;
+import org.elasticsearch.indices.IndicesService;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportActionProxy;
+import org.elasticsearch.transport.TransportResponse;
+import org.elasticsearch.transport.TransportService;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+
+public class GetFileChunkAction extends ActionType<GetFileChunkAction.Response> {
+
+    public static final String NAME = "internal:crate:replication/logical/file_chunk/get";
+    public static final GetFileChunkAction INSTANCE = new GetFileChunkAction();
+
+    public GetFileChunkAction() {
+        super(NAME);
+    }
+
+    @Override
+    public Writeable.Reader<Response> getResponseReader() {
+        return Response::new;
+    }
+
+    @Singleton
+    public static class TransportAction extends TransportSingleShardAction<Request, Response> {
+
+        private final IndicesService indicesService;
+        private final PublisherRestoreService publisherRestoreService;
+
+        @Inject
+        public TransportAction(ThreadPool threadPool,
+                               ClusterService clusterService,
+                               TransportService transportService,
+                               IndicesService indicesService,
+                               PublisherRestoreService publisherRestoreService,
+                               IndexNameExpressionResolver indexNameExpressionResolver) {
+            super(
+                NAME,
+                threadPool,
+                clusterService,
+                transportService,
+                indexNameExpressionResolver,
+                Request::new,
+                ThreadPool.Names.GET
+            );
+            this.indicesService = indicesService;
+            this.publisherRestoreService = publisherRestoreService;
+            TransportActionProxy.registerProxyAction(transportService, NAME, Response::new);
+        }
+
+        @Override
+        protected Response shardOperation(Request request,
+                                          ShardId shardId) throws IOException {
+            logger.debug(request.toString());
```

Comment:
Maybe wrap in `isDebugEnabled` to avoid computing `toString`

Reply:
This was a develop trace I missed to remove, thanks for catching.